api/src/main/java/io/opentelemetry/trace/Tracestate.java

```diff
@@ -235,28 +235,29 @@ public static Entry create(String key, String value) {
 
   // Key is opaque string up to 256 characters printable. It MUST begin with a lowercase letter, and
   // can only contain lowercase letters a-z, digits 0-9, underscores _, dashes -, asterisks *, and
-  // forward slashes /.
+  // forward slashes /.  For multi-tenant vendor scenarios, an at sign (@) can be used to prefix the
+  // vendor name.
   private static boolean validateKey(String key) {
-    if (key.length() > KEY_MAX_SIZE
-        || key.isEmpty()
-        || key.charAt(0) < 'a'
-        || key.charAt(0) > 'z') {
+    if (key.length() > KEY_MAX_SIZE || key.isEmpty() || isNotNumberOrDigit(key.charAt(0))) {
       return false;
     }
+    int atSeenCount = 0;
     for (int i = 1; i < key.length(); i++) {
       char c = key.charAt(i);
-      if (!(c >= 'a' && c <= 'z')
-          && !(c >= '0' && c <= '9')
-          && c != '_'
-          && c != '-'
-          && c != '*'
-          && c != '/') {
+      if (isNotNumberOrDigit(c) && c != '_' && c != '-' && c != '@' && c != '*' && c != '/') {
+        return false;
+      }
+      if ((c == '@') && (++atSeenCount > 1)) {
         return false;
       }
     }
     return true;
   }
 
+  private static boolean isNotNumberOrDigit(char ch) {
+    return (ch < 'a' || ch > 'z') && (ch < '0' || ch > '9');
+  }
```

Comment:
I would normally write a test method like this in the positive sense "isNumberOrDigit()" and apply the logical negation at the usage site. I find it easier to debug that way. YMMV.

Reply:
Yeah, in a moment of weakness, I listened to IDEA and changed it to the negative version, because all usages were negated. Happy to change it back. :)

Reply:
fixed.